### PR TITLE
feat(desktop): zero-config Claude detection + first-run onboarding

### PR DESCRIPTION
**Pull request by softmarshmallow**
## What

Zero-config Claude for new desktop users (#813). On first launch, Grida Desktop now **detects the user's `claude` PATH-independently** and walks them through a short **first-run onboarding** instead of failing silently when Claude isn't set up.

Two halves:

### 1. PATH-independent detection (`@grida/agent`)
- `path-resolve.ts` — resolves `claude`/`node`/`npx` against well-known install dirs (`~/.local/bin`, `/opt/homebrew/bin`, npm-global, version-manager shims, …) merged with `process.env.PATH`. This is the **packaged-GUI fix**: a Finder/Dock-launched app inherits a stripped PATH, so these binaries are otherwise unfindable. Pure, platform-injectable, unit-tested.
- `detect.ts` — `detectClaude()`, a cheap never-throws fs probe (binary presence only; **not** a login check).
- `claude.ts` `spawnBridge` applies the same augmented PATH so detection and the real ACP bridge spawn share one source of truth.
- Wired end-to-end over the existing desktop bridge: `POST /providers/claude/detect` → `transport` → `@grida/desktop-bridge` type → `preload` → `editor/lib/desktop/bridge.ts` (`providers.detectClaude()`), mirroring the existing `providers.probeEndpoint` surface.

### 2. First-run onboarding modal (`editor/scaffolds/desktop/onboarding/`)
- A non-dismissible multi-step wizard: **Welcome → Choose your AI → Open a workspace → You're all set**. Claude detection is **one item** in the AI step (Detected ✓ / Not found + an install hint & re-check), with BYOK / local-model alternatives routing into Settings.
- Gated once per machine via `onboarding-flag.ts` (localStorage), shown from the desktop welcome page. Every non-final step has **Skip** — never a hard wall.
- Full-bleed per-step artwork in the **Grida house style** (constructed geometry + vector anchor-node dots + a purple→teal brand-gradient focal), shipped as **static light/dark SVGs in `/public/onboarding/`** and rendered via plain `` with a `dark:` swap.

## Why

A new user with Claude installed should "just use Grida." Before this, there was no detection and no onboarding — and a packaged GUI launch couldn't even find `claude`, so first runs failed as a generic error. This is the "zero-config auto-detect + default UX" slice of #813 (and shares the PATH-independent discovery with #871 part 2). Deliberately lean per the "feel it first" scope: well-known-dirs PATH (not login-shell capture), binary-presence detection (no authoritative login probe).

## Reviewer notes

- **Commits are sliced** for review: (1) detection backend + spawn fix, (2) bridge wire, (3) onboarding UI + assets.
- **Security (GRIDA-SEC-004):** the new `detect_claude` rides the existing desktop-sidecar boundary — read-only, no new `get`, no boundary widening. The spawn change executes `claude`/`node`/`npx` only from the user's own well-known dirs (same trust as their shell). `next/image` is intentionally avoided under `/desktop` (CSP blocks the optimizer roundtrip) — plain `` with a scoped disable.
- **Verification:** `oxfmt` + `oxlint` clean; editor typecheck passes; the 14 new agent tests (`path-resolve`, `detect`, `providers` route) pass. Onboarding illustrations rendered and inspected in light + dark.
- **Known follow-ups (filed, not in scope here):**
  - #893 — detection shows "ready" but doesn't verify `claude login`, and the result isn't propagated to the composer → a logged-out first run can still fail. **The most important follow-up.**
  - #894 — choosing a non-Claude provider ejects the user out of the onboarding flow.
  - #895 — the bridge spawn strips `ANTHROPIC_API_KEY` unconditionally (subscription-billing decision; edge case for API-key users).

## Test plan

Unit (CI): the new `@grida/agent` tests above.

Live (cannot be asserted in CI):
- Launch the **packaged** app from Finder (the stripped-PATH condition) with `claude` installed in a well-known dir → onboarding shows "Detected", a real run spawns the bridge successfully.
- Fresh profile → onboarding appears; Continue/Skip → never reappears across restart. Reset for testing: `localStorage.removeItem("grida.desktop.onboarding.completed.v1")`.
- Uninstalled `claude` → "Not found" + install hint; "I've installed it" re-detects without restart.

## Summary by CodeRabbit

* **New Features**
  * Added a first-run desktop onboarding flow (welcome, AI selection, folder selection, finish/recap).
  * Added Claude CLI installation detection to the onboarding AI step, including an install prompt and a link to open settings when needed.
* **Bug Fixes**
  * Onboarding completion/dismissal is now persisted to prevent it from reappearing after restart.
  * Desktop welcome now avoids rendering mismatches by enabling onboarding only after mount for persisted state.
  * Older desktop builds handle Claude detection without blocking onboarding.
  * Completing onboarding can refresh and select the newly opened workspace immediately.

### CODE REVIEW

File: editor/scaffolds/desktop/onboarding/steps/workspace-step.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+      setBusy(true);
+      setError(null);
+      // `createDirectory` surfaces the macOS "New Folder" button; macOS-only,
+      // safely ignored elsewhere (mirrors the welcome page).
+      const paths = await bridge.dialog.open({
+        properties: ["openDirectory", "createDirectory"],
+      });
+      if (!paths || paths.length === 0) return;
+      const ws = await workspacesNs.openFolder(paths[0]);
+      update({ openedWorkspace: ws });
```
[the remark this answers is not in the document]
Comment (softmarshmallow) [reply editor/scaffolds/desktop/onboarding/steps/workspace-step.tsx:41]: Fixed — `FirstRunOnboarding` now passes the workspace opened mid-flow back through `onDone(openedWorkspaceId)`, and the welcome page runs `refreshWorkspaces()` + selects it on close, so the composer and recents reflect it immediately rather than only on the next focus event.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: editor/app/desktop/welcome/page.tsx [hunk 2]
```diff
@@ -69,6 +70,12 @@ export default function DesktopWelcomePage() {
   const [busy, setBusy] = useState(false);
   const [error, setError] = useState<string | null>(null);
   const [pickerOpen, setPickerOpen] = useState(false);
+  // First-run onboarding (issue #813): zero-config Claude detection. Lazily
+  // seeded from the persisted flag so a returning user never sees it; the
+  // localStorage read is client-only, which is fine on this desktop-only page.
+  const [onboarding, setOnboarding] = useState(
+    () => !onboarding_flag.isComplete()
+  );
```
[the remark this answers is not in the document]
Comment (softmarshmallow) [reply editor/app/desktop/welcome/page.tsx]: Fixed — onboarding visibility is now decided in a mount `useEffect` instead of the `useState` initializer, so the server render and first client render both start hidden (no client-only `localStorage` read during SSR → no hydration mismatch or modal flash for returning users).

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: packages/grida-ai-agent/src/agent-provider/path-resolve.test.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    expect(dirs).toContain(path.join(HOME, ".local", "bin"));
+  });
+});
+
+describe("claude_path.augmentedPathValue", () => {
+  it("joins the dirs with the platform delimiter", () => {
+    const value = claude_path.augmentedPathValue(
+      posix(["/opt/homebrew/bin"], { PATH: "/usr/bin" })
+    );
+    expect(value.split(path.delimiter)).toContain("/opt/homebrew/bin");
```
[the remark this answers is not in the document]
Comment (softmarshmallow) [reply packages/grida-ai-agent/src/agent-provider/path-resolve.test.ts]: Fixed — the assertion now splits on `path.posix.delimiter` to match the mocked `linux` platform, so it no longer depends on the host OS (would have broken on the Windows runner).

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
